tests/integration/targets/python_requirements_info/tasks/731-parse-project-extra-syntax.yml

```
@@ -0,0 +1,59 @@
+---
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Install Python sampleprojectpy2 libraries
+  pip:
+    name: sampleprojectpy2[test]>=1.3.1
+
+- name: Ensure sampleprojectpy2 is installed
+  python_requirements_info:
+    dependencies: sampleprojectpy2[test]>=1.3.1
+  register: _py_facts
+  failed_when: | 
+      (_py_facts.msg is defined) and (_py_facts.msg | regex_search("Failed to parse")) or
+      (_py_facts.valid is not defined) or
+      ('sampleprojectpy2[test]' not in _py_facts.valid)
+
```

Now the tricky thing is, if we start from scratch (nothing has been installed yet) and run it like:
```suggestion
- name: Install Python sampleprojectpy2 libraries
  pip:
    name: sampleprojectpy2>=1.3.1

- name: Ensure sampleprojectpy2 is installed
  python_requirements_info:
    dependencies: sampleprojectpy2[test]>=1.3.1
  register: _py_facts
  failed_when: | 
      (_py_facts.msg is defined) and (_py_facts.msg | regex_search("Failed to parse")) or
      (_py_facts.valid is not defined) or
      ('sampleprojectpy2[test]' not in _py_facts.valid)
```
It should fail, because `[test]` was not used in the installation, but will it? From the code, I think we are not checking whether the extras are installed or not.

---

You're right. This is not failing.

---

And there is today no way to do that. I'll propose another approach.